src/components/Homepage.jsx

```diff
@@ -1,5 +1,51 @@
+import { useContext, useState } from 'react';
+import { UserContext } from '../contexts/userContext';
+import { getUsers } from '../utils/utils';
+
 const Homepage = () => {
-  return <h2>This is the homepage</h2>;
+  const { user, setUser } = useContext(UserContext);
+
+  const [input, setInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState(false);
+
+  const inputHandler = event => {
+    setInput(event.target.value);
+    setErrorMessage(false);
+  };
+
+  const submitHandler = event => {
+    event.preventDefault();
+    getUsers().then(({ users }) => {
+      const username = users.map(userData => {
+        return userData.username;
+      });
+      if (username.includes(input)) {
+        setUser(input);
+      } else {
+        setErrorMessage(true);
+      }
+    });
+    setInput('');
+  };
+
+  return (
+    <>
+      <h2>Welcome{user !== '' ? ` ${user}` : ''}!</h2>
+      <form onSubmit={submitHandler}>
+        <label htmlFor="username">Username</label>
+        <input
+          type="text"
+          name="username"
+          id="username"
+          placeholder="your username"
+          onChange={inputHandler}
+          value={input}
+        />
+        <p>{errorMessage ? "user doesn't exist" : null}</p>
+        <button>Login</button>
+      </form>
```

Comment:
I like that you have this login functionality, though I would bear in mind that someone wanting to try this out without knowing what the correct usernames might have a hard time using this. You could provide them with a list of usernames to choose from, or have them logged in as a valid user by default